app/src/main/java/org/mozilla/focus/web/GeckoWebViewProvider.kt

```diff
@@ -0,0 +1,598 @@
+            return webViewTitle
+        }
+
+        override fun exitFullscreen() {
+            geckoSession.exitFullScreen()
+        }
+
+        override fun findAllAsync(find: String) {
+            finder.find(find, 0).then({ result ->
+                if (result != null) {
+                    findListener?.onFindResultReceived(result.current, result.total, true)
+                }
+                GeckoResult<Void>()
+            }, { _ ->
+                GeckoResult<Void>()
+            })
+        }
+
+        override fun findNext(forward: Boolean) {
+            finder.find(null, if (forward) 0 else GeckoSession.FINDER_FIND_BACKWARDS)
+                .then({ result ->
+                    if (result != null) {
+                        findListener?.onFindResultReceived(result.current, result.total, true)
+                    }
+                    GeckoResult<Void>()
+                }, { _ ->
+                    GeckoResult<Void>()
+                })
+        }
+
+        override fun clearMatches() {
+            finder.clear()
+        }
+
+        override fun setFindListener(findListener: IFindListener) {
+            this.findListener = findListener
+        }
+
+        override fun loadData(
+            baseURL: String,
+            data: String,
+            mimeType: String,
+            encoding: String,
+            historyURL: String
+        ) {
+            geckoSession.loadData(data.toByteArray(Charsets.UTF_8), mimeType, baseURL)
+            currentUrl = baseURL
+            isLoadingInternalUrl = currentUrl == LocalizedContent.URL_RIGHTS || currentUrl ==
+                    LocalizedContent.URL_ABOUT
+        }
+
+        private fun sendTelemetrySnapshots() {
+            val response = GeckoResponse<GeckoBundle> { value ->
+                if (value != null) {
+                    try {
+                        val jsonData = value.toJSONObject()
+                        TelemetryWrapper.addMobileMetricsPing(jsonData)
+                    } catch (e: JSONException) {
+                        Log.e("getSnapshots failed", e.message)
+                    }
+                }
+            }
+
+            geckoRuntime!!.telemetry.getSnapshots(true, response)
+        }
+
+        override fun onDetachedFromWindow() {
+            PreferenceManager.getDefaultSharedPreferences(context)
+                .unregisterOnSharedPreferenceChangeListener(this)
+            super.onDetachedFromWindow()
+        }
+
+        companion object {
+            private const val TAG = "GeckoWebView"
+        }
+    }
+
+    companion object {
+        @Volatile
+        private var geckoRuntime: GeckoRuntime? = null
+        private var internalAboutData: String? = null
+        private var internalRightsData: String? = null
+        private const val firstProgress = 10
+        private const val quarterProgress = 25
+        private const val finalProgress = 100
+        private const val userAgent =
```

Comment:
\[detekt\] INFO: \[style/NamingConventionViolation\] All names in the codebase should be matching the naming convention of the codebase\.